File: 이것이 취업을 위한 코딩 테스트다/그리디/큰_수의_법칙/큰_수의_법칙.cpp
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int n, m, k;
vector<int> v;

int main() {
  cin >> n >> m >> k;

  for (int i = 0; i < n; i++) {
    int x;
    cin >> x;
    v.push_back(x);
  }

  sort(v.begin(), v.end(), greater <>());

  int first = v[0], second = v[1];
  int answer = 0;

  for (int i = 1; i <= m; i++) {
    if (i % (k + 1) == 0) {
      answer += second;
    } else {
      answer += first;
    }
  }

  cout << answer;
}

File: 이것이 취업을 위한 코딩 테스트다/그리디/큰_수의_법칙/큰_수의_법칙.java
package 그리디.큰_수의_법칙;

import java.util.*;

public class 큰_수의_법칙 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    int n, m, k;

    n = sc.nextInt();
    m = sc.nextInt();
    k = sc.nextInt();

    int arr[] = new int[n];

    for (int i = 0; i < n; i++) {
      arr[i] = sc.nextInt();
    }

    Arrays.sort(arr);
    int first = arr[n - 1];
    int second = arr[n - 2];
    int answer = 0;

    for (int i = 1; i <= m; i++) {
      if (i % (k + 1) == 0) {
        answer += second;
      } else {
        answer += first;
      }
    }

    System.out.println(answer);
  }
}
